### DISCUSSION TITLE
How to discover the guild instead of implicitly setting it in code

### DISCUSSION THREAD

**Discussion by tsal** [serenity-rs/serenity#2168] [Q&A] [answered]
Hey all! I'm very new to Discord development, and a novice (in professional terms) with Rust, so bear with me if I've missed something obvious.

The examples all seem to require a Guild ID to be set, but I do not want my bot limited to one guild. Is there a way to get that information from the `Ready` struct somehow when I'm implementing `ready()` on the handler? I see there's a vector of guilds, but I'm not sure what I need to "pick" the right one for the slash command response.

**Comment by tsal** [comment, marked as answer]
.. I seem to have maybe found an answer in that I may be asking the wrong question or misunderstanding the example.. should I just loop through them all in `ready()` and register them for all available guilds? I don't want to get myself banned from the API, hence why I'm hesitating.

EDIT: See below response from @acdenisSK - I missed that there is a global version of `set_application_commands`.

**Reply by arqunis** [reply]
If you wish to register commands for all guilds, just use [`Command::set_global_application_commands`](https://serenity-rs.github.io/serenity/current/serenity/model/application/command/struct.Command.html#method.set_global_application_commands). No need to loop through all guilds; the only reason we use guilds in the example is because it's faster that way (registering commands in a guild is almost instant) and allows users to test commands in a quick and private manner before 'publishing' them to a wider audience (i.e. every guild and in DMs).

**Reply by tsal** [reply]
As I was digging more to add more detail to what I was trying to do, I saw `set_global_application_commands`. I came here to update and mark this answered, but you were way too fast @acdenisSK -- thank you!!
